### PR TITLE
fix(dashmate): dashmate helper crashing

### PR DESCRIPTION
**Pull request by shuplenkov**
## Issue being fixed or feature implemented

Dashmate helper crashes if docker container running in linux environment.

The If conditions below didn't work correctly and always led to `then` condition. And it worked on Mac, but didn't work on Linux.

`if [ ! $(getent passwd $USER_ID | grep $USER_ID -q) ]`
`if [ -z $(getent group $DOCKER_GROUP_ID) ] ; then`
`if [ ! $(id -nG $USERNAME | grep -q $GROUP) ]; then` 

## What was done?

Fixed dashmate helper `entrypoint.sh` script conditions. 

## How Has This Been Tested?

Manual

## Breaking Changes

No

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/dashmate/docker/entrypoint.sh [hunk 1]
```diff
@@ -6,25 +6,26 @@ USERNAME=dashmate
 GROUP=docker
 
 # check if user with our uid exists in the system
-if [ ! $(getent passwd $USER_ID | grep $USER_ID -q) ]; then
+if ! (getent passwd $USER_ID | grep -q $USER_ID); then
   echo "Creating user $USERNAME ($USER_ID)"
```
Comment (shumkov) [inline packages/dashmate/docker/entrypoint.sh:10]: Why is username is important? Can we use own username with same ID inside docker container and just rewrite DASHMATE_HOME_DIR for helper. So we don't need to deal with symlinks.

Comment (shuplenkov) [reply packages/dashmate/docker/entrypoint.sh:10]: In Ubuntu you will get this - `adduser: uid '1000' in use`

Comment (shumkov) [reply packages/dashmate/docker/entrypoint.sh:10]: Why do we create a user with an existing ID?

File: packages/dashmate/src/docker/DockerCompose.js [hunk 2]
```diff
@@ -417,6 +417,7 @@ class DockerCompose {
       DASHMATE_HOME_DIR: HOME_DIR_PATH,
       LOCAL_UID: uid,
       LOCAL_GID: gid,
+      DOCKER_GID: 998, // TODO: on Linux: `cat /etc/group|grep ^docker: |cut -d: -f3`, on Mac: set to 0
```
Comment (shumkov) [inline packages/dashmate/src/docker/DockerCompose.js:420]: getent group docker ?

File: packages/dashmate/src/config/ensureHomeDirFactory.js [hunk 3]
```diff
@@ -27,6 +27,7 @@ function ensureHomeDirFactory() {
 
     try {
       fs.mkdirSync(HOME_DIR_PATH);
+      fs.chmodSync(HOME_DIR_PATH, 0o700);
```
Comment (pshenmic) [inline packages/dashmate/src/config/ensureHomeDirFactory.js:30]: Why do we need it?

**Review by pshenmic — APPROVED** [review]

File: packages/dashmate/src/config/ensureHomeDirFactory.js (hunk 3 above)
Comment (pshenmic) [inline packages/dashmate/src/config/ensureHomeDirFactory.js:30]: Why do we need it?

**Review by pshenmic — CHANGES_REQUESTED** [review]

**Review by shumkov — APPROVED** [review]
👍

**Review by pshenmic — APPROVED** [review]
Proved to be working on mac, ubuntu (22.04), and devnet

**Pull request merged**